File: src/main/java/org/books/WebMvcConfig.java
package org.books;

import org.books.interceptors.AuthorizationInterceptor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebMvcConfig implements WebMvcConfigurer {

    @Override public void addInterceptors(InterceptorRegistry registry) {
        registry
            .addInterceptor(new AuthorizationInterceptor())
            .excludePathPatterns("/api/public");
    }
}

File: src/main/java/org/books/WebMvcConfig.kt
package org.books

import org.books.interceptors.AuthorizationInterceptor
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebMvcConfig : WebMvcConfigurer {

    @Bean
    fun authorizationInterceptor() : AuthorizationInterceptor {
        return AuthorizationInterceptor()
    }

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry!!
                .addInterceptor(authorizationInterceptor())
                .excludePathPatterns("/api/public")
    }

}
